src/runtime/typemanager.cs

```diff
@@ -422,6 +434,198 @@ internal static IntPtr BasicSubType(string name, IntPtr base_, Type impl)
+            internal static int nb_multiply = 29;
+            internal static int nb_negative = 30;
+            internal static int nb_or = 31;
+            internal static int nb_positive = 32;
+            internal static int nb_power = 33;
+            internal static int nb_remainder = 34;
+            internal static int nb_rshift = 35;
+            internal static int nb_subtract = 36;
+            internal static int nb_true_divide = 37;
+            internal static int nb_xor = 38;
+            internal static int sq_ass_item = 39;
+            internal static int sq_concat = 40;
+            internal static int sq_contains = 41;
+            internal static int sq_inplace_concat = 42;
+            internal static int sq_inplace_repeat = 43;
+            internal static int sq_item = 44;
+            internal static int sq_length = 45;
+            internal static int sq_repeat = 46;
+            internal static int tp_alloc = 47;
+            internal static int tp_base = 48;
+            internal static int tp_bases = 49;
+            internal static int tp_call = 50;
+            internal static int tp_clear = 51;
+            internal static int tp_dealloc = 52;
+            internal static int tp_del = 53;
+            internal static int tp_descr_get = 54;
+            internal static int tp_descr_set = 55;
+            internal static int tp_doc = 56;
+            internal static int tp_getattr = 57;
+            internal static int tp_getattro = 58;
+            internal static int tp_hash = 59;
+            internal static int tp_init = 60;
+            internal static int tp_is_gc = 61;
+            internal static int tp_iter = 62;
+            internal static int tp_iternext = 63;
+            internal static int tp_methods = 64;
+            internal static int tp_new = 65;
+            internal static int tp_repr = 66;
+            internal static int tp_richcompare = 67;
+            internal static int tp_setattr = 68;
+            internal static int tp_setattro = 69;
+            internal static int tp_str = 70;
+            internal static int tp_traverse = 71;
+            internal static int tp_members = 72;
+            internal static int tp_getset = 73;
+            internal static int tp_free = 74;
+            internal static int nb_matrix_multiply = 75;
+            internal static int nb_inplace_matrix_multiply = 76;
+            internal static int am_await = 77;
+            internal static int am_aiter = 78;
+            internal static int am_anext = 79;
+            internal static int tp_finalize = 80;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct PY_TYPE_SLOT
+        {
+            public long slot; //slot id, from typeslots.h
+            public IntPtr func; //function pointer of the function implementing the slot
+        }
+
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+        internal class PyTypeSpecOffset
```

Comment:
Ideally, these extra types should go into their own files.

Reply:
@lostmsu I intend to do it after I get things working as part of the final cleanup